### PR TITLE
refactor(bigtable): connection refresh into own file

### PR DESCRIPTION
**Pull request by dbolduc**
**Context:** ~1 yr from now `CommonClient<>` gets deleted. But we are going to keep `ConnectionRefreshState` for use by the Stubs. So we might as well factor it out now.

The only manual changes were to the `#include`s and copyright years.

<hr />

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8831)

### CODE REVIEW

File: google/cloud/bigtable/internal/connection_refresh_state.h [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  // Cancel all currently registered timers and all which will be registered in
+  // the future.
+  void CancelAll();
+
+ private:
+  void DeregisterTimer(std::uint64_t id);
+  std::mutex mu_;
+  bool shutdown_ = false;           // GUARDED_BY(mu_)
+  std::uint64_t id_generator_ = 0;  // GUARDED_BY(mu_)
+  std::unordered_map<std::uint64_t,
```
Comment (coryan) [inline google/cloud/bigtable/internal/connection_refresh_state.h:57]: `#include <unordered_map>`

File: google/cloud/bigtable/internal/connection_refresh_state.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  void RegisterTimer(future<void> fut);
+  // Cancel all currently registered timers and all which will be registered in
+  // the future.
+  void CancelAll();
+
+ private:
+  void DeregisterTimer(std::uint64_t id);
+  std::mutex mu_;
+  bool shutdown_ = false;           // GUARDED_BY(mu_)
+  std::uint64_t id_generator_ = 0;  // GUARDED_BY(mu_)
```
Comment (coryan) [inline google/cloud/bigtable/internal/connection_refresh_state.h:56]: nitty nit: `#include <cstdint>`

File: google/cloud/bigtable/internal/connection_refresh_state.h [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+namespace internal {
+
+/**
+ * Time after which we bail out waiting for a connection to become ready.
+ *
+ * This number was copied from the Java client and there doesn't seem to be a
+ * well-founded reason for it to be exactly this. It should not bee too large
+ * since waiting for a connection to become ready is not cancellable.
+ */
+std::chrono::seconds constexpr kConnectionReadyTimeout(10);
```
Comment (coryan) [inline google/cloud/bigtable/internal/connection_refresh_state.h:39]: More often than not we write:

```cc
auto constexpr kConnectionReadyTimeout = std::chrono::seconds(10);
```

Does this need to be in the header?

**Review by coryan — APPROVED** [review]
A few nits, please fix and then merge

File: google/cloud/bigtable/internal/connection_refresh_state.h (hunk 3 above)
Comment (dbolduc) [reply google/cloud/bigtable/internal/connection_refresh_state.h]: Moved.

File: google/cloud/bigtable/internal/connection_refresh_state.h (hunk 2 above)
Comment (dbolduc) [reply google/cloud/bigtable/internal/connection_refresh_state.h:49]: Done.

File: google/cloud/bigtable/internal/connection_refresh_state.h (hunk 1 above)
Comment (dbolduc) [reply google/cloud/bigtable/internal/connection_refresh_state.h:50]: Done.

**Pull request merged**